pkg/solana/capabilities/targets/write_target.go

```diff
@@ -0,0 +1,214 @@
+// 		case string:
+// 			val, err := pipeline.VarExpr(v, vars)()
+// 			if err == nil {
+// 				args = append(args, val)
+// 			} else if errors.Is(errors.Cause(err), pipeline.ErrParameterEmpty) {
+// 				args = append(args, param)
+// 			} else {
+// 				return args, err
+// 			}
+// 		default:
+// 			args = append(args, param)
+// 		}
+// 	}
+
+// 	return args, nil
+// }
+
+func (cap *SolanaWrite) Execute(ctx context.Context, callback chan<- capabilities.CapabilityResponse, request capabilities.CapabilityRequest) error {
+	cap.lggr.Debugw("Execute", "request", request)
+
+	config := cap.chain.Config().ChainWriter()
+
+	if config == nil {
+		return fmt.Errorf("ChainWriter config undefined")
+	}
+
+	reqConfig, err := parseConfig(request.Config)
+	if err != nil {
+		return err
+	}
+
+	inputsAny, err := request.Inputs.Unwrap()
+	if err != nil {
+		return err
+	}
+	inputs := inputsAny.(map[string]any)
+
+	blockhash, err := cap.reader.LatestBlockhash()
+
+	programID := config.ForwarderProgramID
+	state := config.ForwarderStateAccount
+	authority := config.FromAddress
+
+	// Determine store authority
+	// TODO: compute this only once per capability init
+	seeds := [][]byte{[]byte("forwarder"), state.Bytes()}
+	forwarderAuthority, _, err := sdk.FindProgramAddress(seeds, programID)
+	if err != nil {
+		return errors.Wrap(err, "error on FindProgramAddress")
+	}
+
+	data := []byte{}
+
+	// No signature validation in the MVP demo
+	signatures := [][]byte{} // TODO: validate each sig is 64 bytes
+
+	data = append(data, uint8(len(signatures))) // length prefix
+	for _, sig := range signatures {
+		data = append(data, sig...)
+	}
+
+	// TODO: encode inputs into data
+	data = append(data, inputs["report"].([]byte)...)
+
+	accounts := []*sdk.AccountMeta{
+		// state
+		{PublicKey: state, IsWritable: false, IsSigner: false},
+		// authority (node's transmitter key that's the tx signer)
+		{PublicKey: authority, IsWritable: false, IsSigner: true},
+		// forwarder_authority
+		{PublicKey: forwarderAuthority, IsWritable: false, IsSigner: false},
+		// receiver_program
+		{PublicKey: reqConfig.ReceiverProgramID, IsWritable: false, IsSigner: false},
+		// ... rest passed from reqConfig
+	}
+
+	for _, account := range reqConfig.Accounts {
+		publicKey, err := sdk.PublicKeyFromBase58(account.PublicKey)
+		if err != nil {
+			return err
+		}
+		accounts = append(accounts, &sdk.AccountMeta{
+			PublicKey:  publicKey,
+			IsWritable: account.IsWritable,
+			IsSigner:   account.IsSigner,
+		})
+	}
+
+	tx, err := sdk.NewTransaction(
+		[]sdk.Instruction{
+			sdk.NewInstruction(config.ForwarderProgramID, accounts, data),
+		},
+		blockhash.Value.Blockhash,
+		sdk.TransactionPayer(authority),
+	)
+	if err != nil {
+		return errors.Wrap(err, "error on Transmit.NewTransaction")
+	}
+
+	if err = cap.chain.TxManager().Enqueue(authority.String(), tx); err != nil {
+		return err
+	}
+
+	go func() {
+		// TODO: cast tx.Error to Err (or Value to Value?)
```

Comment:
In this case, we are just returning any errors and no values right? I'm assuming the solana manager doesn't have any sort of transaction receipt or a blockhash to confirm that the transaction has finalized.